pkg/loki/modules.go

```diff
@@ -437,6 +437,9 @@ func (t *Loki) initStore() (_ services.Service, err error) {
 			// Use AsyncStore to query both ingesters local store and chunk store for store queries.
 			// Only queriers should use the AsyncStore, it should never be used in ingesters.
 			asyncStore = true
+		case t.Cfg.isModuleEnabled(IndexGateway):
+			// we want to use the actual storage when running the index-gateway, so we remove the Addr from the config
+			t.Cfg.StorageConfig.BoltDBShipperConfig.IndexGatewayClientConfig.Address = ""
```

Comment:
I didn't get this part - doesn't it basically means that we can't run boltdbshipper with an Index Gateway that has no ring?

Reply:
It is just clearing the address so that `NewIndexClient` initializes `boltdb-shipper` index client instead of index gateway client.

Reply:
Basically we do not want to initialize index gateway client from index gateway.

Reply:
ohhhh got it